### DISCUSSION TITLE
Getting error "name 'dynamic' is not defined" on saspy.SASsession

### DISCUSSION THREAD

**Discussion by wailoonho** [sassoftware/saspy#588] [General]
I'm new to Python and SAS. 
I want to develop sas etl using python, but i still having problem connecting to sas from python.

sascfg_personal.py
`
saspy.SASsession

staging = {
    'iomhost' : 'unastgsascmp01.hq.tnb.com.my',
    'iomport' : 8591,
    'provider': 'sas.iomprovider',
    'encoding': 'utf-8'}
`

prog.py
`
import saspy

sas = saspy.SASsession(cfgname='staging')
`

Error "name 'dynamic' is not defined" is on the line of sas=saspy.SASsession.

I already checked, dynamic is use for old saspy version, but mine is 5.6.0.
May I know what is the actual issue and how should I solve?

**Comment by tomweber-sas** [comment]
You're trying to use the COM access method. It requires other packages to be able to be used. The doc is [here](https://sassoftware.github.io/saspy/configuration.html#com). Also, it's not the recommended access method to use. You should use the IOM access method instead. It is supported and doesn't require any other packages, though it does need java, which is likely already installed on your client. The doc for that is [here](https://sassoftware.github.io/saspy/configuration.html#iom).

Tom
